source/elements/oneDAL/include/oneapi/dal/kmeans.hpp

```diff
@@ -0,0 +1,256 @@
+#include <cstdint>
+#include "oneapi/dal/table.hpp"
+#include "oneapi/dal/train.hpp"
+
+namespace oneapi::dal::kmeans {
+
+namespace method {
+   /// Tag-type that denotes `Lloyd's <kmeans_t_math_lloyd_>`_ computational
+   /// method.
+   struct lloyd {};
+
+   /// Alias tag-type for `Lloyd's <kmeans_t_math_lloyd_>`_ computational
+   /// method.
+   using by_default = lloyd;
+} // namespace method
+
+namespace task {
+   /// Tag-type that parametrizes entities used for solving
+   /// :capterm:`clustering problem <clustering>` .
+   struct clustering {};
+
+   /// Alias tag-type for custering task.
+   using by_default = clustering;
+} // namespace task
+
+/// @tparam Float  The floating-point type that the algorithm uses for
+///                intermediate computations. Can be :expr:`float` or
+///                :expr:`double`.
+/// @tparam Method Tag-type that specifies an implementation of algorithm. Can
+///                be :expr:`method::lloyd`.
+/// @tparam Task   Tag-type that specifies type of the problem to solve. Can
```

Comment:
```suggestion
/// @tparam Task   Tag-type that specifies the type of the problem to solve. Can
```

apply this everywhere else as well